### PR TITLE
#77: Draft of typealias support

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+/** A generated typealias declaration */
+class TypeAliasSpec private constructor(builder: TypeAliasSpec.Builder) {
+  val name: String
+  val type: TypeName
+  val modifiers: Set<KModifier>
+
+  init {
+    this.name = builder.name
+    this.type = builder.type
+    this.modifiers = builder.modifiers
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt]: These can be initialized inline on the property

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt]: Also this is missing `toImmutableSet`

File: src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
```
Comment (JakeWharton) [inline src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt]: 2017

File: src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt]: 2017

File: src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    val typeAliasSpec = TypeAliasSpec
+        .builder("Word", String::class)
+        .visibility(KModifier.PRIVATE)
+        .build()
+    assertThat(typeAliasSpec.toString()).isEqualTo("""
+        |private typealias Word = java.lang.String
+        |""".trimMargin())
+  }
+
+  @Test fun forbiddenModifiers() {
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt]: I think this test is probably overkill. We want our tests to exercise the interesting logic in our code, but not to re-implement that logic

**Comment by swankjesse** [comment]
Supercool!

File: src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt (hunk 4 above)
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/TypeAliasSpecTest.kt]: That makes sense, removed

**Comment by Egorand** [comment]
Oops, looks like `CodeWriter` APIs have changed, will rebase and fix later.

**Comment by swankjesse** [comment]
Thanks!

**Pull request merged**